bpf/lib/trace.h

```diff
@@ -110,6 +103,19 @@ update_trace_metrics(struct __ctx_buff *ctx, __u8 obs_point, enum trace_reason r
 			update_metrics(ctx_full_len(ctx), METRIC_INGRESS,
 				       REASON_DECRYPT);
 		break;
+	/* TRACE_FROM_LXC, i.e endpoint-to-endpoint delivery is handled
+	 * separately in ipv*_local_delivery() where we can bump an egress
+	 * forward. It could still be dropped but it would show up later as an
+	 * ingress drop, in that scenario.
+	 *
+	 * TRACE_{FROM,TO}_PROXY are not handled in datapath. This is because
+	 * we have separate L7 proxy "forwarded" and "dropped" (ingress/egress)
+	 * counters in the proxy layer to capture these metrics.
+	 */
+	case TRACE_FROM_LXC:
```

Comment:
Why add the cases explicitly?

Reply:
Good question. That's one of the upside of using an enumeration for `obs_point`. As soon as we do that, LLVM is able to detect that several switch cases are missing:
```
error: enumeration values 'TRACE_TO_PROXY', 'TRACE_FROM_LXC', and 'TRACE_FROM_PROXY' not handled in switch [-Werror,-Wswitch]
        switch (obs_point) {
                ^
1 error generated.
```
This warning (`-Wswitch`) is enabled by default and turned into an error because we use `-Werror`.

I'll add a note in the commit description. I need to rebase anyway.

Reply:
I've moved this code change to the next commit, as it seemed more appropriate. Thanks for raising this Nathan! :pray: